iothub/device/src/DeviceClient.cs

```diff
@@ -613,12 +617,45 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         /// <summary>
         /// Releases the unmanaged resources used by the DeviceClient and optionally disposes of the managed resources.
         /// </summary>
+        /// <remarks>
+        /// The method <see cref="CloseAsync()"/> should be called before disposing.
+        /// </remarks>
         public void Dispose()
         {
             Dispose(true);
             GC.SuppressFinalize(this);
         }
 
```

Comment:
Should this block of code not be under #if !NET451 || NET472 || NETSTANDARD2_0 ?

Reply:
It doesn't hurt to have it there. It is just a method someone could call, even if they don't have .NET 5.0. The fact it comes from IAsyncDisposable interface, which is only supported in .NET 5.0 is what matters most.

That said, I don't have a problem wrapping it if people prefer that.

Reply:
I've added it, mainly because the doc comments fail to resolve the type.